inception-ui-external-search/src/main/java/de/tudarmstadt/ukp/inception/app/ui/externalsearch/sidebar/ExternalSearchAnnotationSidebar.java

```diff
@@ -0,0 +1,427 @@
+            {
+                private static final long serialVersionUID = 1L;
+
+                {
+                    setChoiceRenderer(new ChoiceRenderer<DocumentRepository>("name"));
+                    setNullValid(false);
+                }
+
+                @Override protected CharSequence getDefaultChoice(String aSelectedValue)
+                {
+                    return "";
+                }
+            };
+            // Just update the selection
+            repositoryCombo.add(new LambdaAjaxFormComponentUpdatingBehavior("change"));
+            add(repositoryCombo);
+
+        }
+
+        private static final long serialVersionUID = -1L;
+    }
+
+    private class SearchForm
+        extends Form<Void>
+    {
+
+        public SearchForm(String id)
+        {
+            super(id);
+            add(new TextField<>("queryInput", searchStateModel.getObject().getTargetQuery(),
+                String.class));
+            LambdaAjaxSubmitLink searchLink = new LambdaAjaxSubmitLink("submitSearch",
+                this::actionSearch);
+            add(searchLink);
+            setDefaultButton(searchLink);
+        }
+
+        private void actionSearch(AjaxRequestTarget aTarget, Form aForm)
+        {
+            IModel<String> targetQuery = searchStateModel.getObject().getTargetQuery();
+            selectedResult = null;
+            if (targetQuery.getObject() == null) {
+                targetQuery.setObject("*.*");
+            }
+
+            searchDocuments(targetQuery.getObject());
+
+            searchStateModel.getObject().getDataProvider().searchDocuments(targetQuery.getObject());
+
+            aTarget.add(dataTableContainer);
+        }
+    }
+
+    private void searchDocuments(String aQuery)
+    {
+        results.clear();
+        applicationEventPublisher.get().publishEvent(new ExternalSearchQueryEvent(this,
+            searchStateModel.getObject().getCurrentRepository().getProject(),
+            userRepository.getCurrentUser().getUsername(), aQuery));
+
+        try {
+            for (ExternalSearchResult result : externalSearchService
+                .query(userRepository.getCurrentUser(),
+                    searchStateModel.getObject().getCurrentRepository(), aQuery)) {
+                results.add(result);
+            }
+        }
+        catch (Exception e) {
+            LOG.error("Unable to perform query", e);
+            error("Unable to load data: " + ExceptionUtils.getRootCauseMessage(e));
+        }
+    }
+
+    public class ResultRowView
+        extends Panel
+    {
+        public ResultRowView(String id, long rowNumber, IModel<ExternalSearchResult> model)
+        {
+            super(id, model);
+
+            ExternalSearchResult result = (ExternalSearchResult) getDefaultModelObject();
+
+            String documentTitle = result.getDocumentTitle();
+
+            Whitelist wl = new Whitelist();
```

Comment:
This code is copied from the external search page. Would be great to factor it out into a utility class so it can be re-used between the sidebar and the external search page.

Reply:
Pulled out the code into a utility class.